RoboJackets/urc-drone
# Feat/aruco gps

# Description
This PR does the following:
- Adds node that detects ArUco markers being used at URC from a camera feed.
- Adds GPS and IMU sensors on the drone.
- Adds node that calculates the latitude and longitude of detected ArUco markers.

Fixes #17 

# Testing Steps
## Test Case 1
1.  Open terminal, cd colcon-urc --> colcon build --> . install/setup.bash --> ros2 launch urc_gazebo simulation.launch.py
2.  Second terminal, cd colcon-urc --> colcon build --> . install/setup.bash --> ros2 launch urc_aruco aruco_detector.launch.py
3.  Third terminal, cd colcon-urc --> . install/setup.bash --> ros2 topic echo /aruco
4.  Fourth terminal, cd colcon-urc --> . install/setup.bash --> ros2 topic echo /aruco_location/tag_location
5.  Rotate Aruco marker in gazebo world to be fully visible in camera frame
6.  Verify publishing aruco tag angles, distance, id, latitude, and longitude


## Test Case 2
1.  Go to urc_gazebo/urdf/worlds/flat_world.world
2. Uncomment Duplicate marker 0 model
3. Follow all steps from Test Case 1, verify adding second tag does not change data from original (or first) tag detected by camera

## Test Case 3
1. Go to urc_gazebo/urdf/worlds/flat_world.world
2. Uncomment Extra tag marker models
3. Follow all steps from Test Case 1, verify detector can detect all 6 ArUco tags


Expectation: Detector detects all 6 unique Aruco markers and rejects duplicate tags that may appear in camera frame

# Self Checklist
